explainers/subresource-loading-opaque-origin-iframes.md

```diff
@@ -0,0 +1,99 @@
+# Subresource loading with Web Bundles: Support opaque origin iframes
+
+Last updated: Apr 2021
+
+This is an extension to [Subresource loading with Web Bundles]. This extension
+allows a bundle to include `urn:uuid:` URL resources, which will be used to
+create an opaque origin iframe.
+
+## Goals
+
+Support the use case of
+[WebBundles for Ad Serving](https://github.com/WICG/webpackage/issues/624).
+
+## Extension to [Subresource loading with Web Bundles]
+
+In this section, _the explainer_ means the [Subresource loading with Web
+Bundles] explainer.
+
+### Allow `urn:uuid:` resources
+
+In addition to the same origin subresource explained in the
+[`<link>`-based API](https://github.com/WICG/webpackage/blob/main/explainers/subresource-loading.md#link-based-api)
+section in the explainer, this extension allows a bundle to include a
+[`urn:uuid:`](https://tools.ietf.org/html/rfc4122) URL subresource.
+
+### Opaque origin iframes
+
+If a `<iframe>`'s `src` attribute is a `urn:uuid:` URL subresource in the
+bundle, the iframe must be instantiated as an
+[opaque origin](https://html.spec.whatwg.org/multipage/origin.html#concept-origin-opaque)
+iframe.
+
+## Example
+
+### The bundle
+
+Suppose that the bundle, `subresources.wbn`, includes the following resources:
+
+```
+- urn:uuid:f81d4fae-7dec-11d0-a765-00a0c91e6bf6
+- … (omitted)
+```
+
+### The main document
+
+```html
+<linkn
+  rel="webbundle"
+  href="https://example.com/dir/subresources.wbn"
+  resources="urn:uuid:f81d4fae-7dec-11d0-a765-00a0c91e6bf6"
+/>
+
+<iframe src="urn:uuid:f81d4fae-7dec-11d0-a765-00a0c91e6bf6"></iframe>
+```
+
+`urn:uuid:f81d4fae-7dec-11d0-a765-00a0c91e6bf6` is loaded from the bundle, and a
+subframe is instantiated as an
+[opaque origin](https://html.spec.whatwg.org/multipage/origin.html#concept-origin-opaque)
+iframe.
+
+Note:
+
+- `url:uuid:` resource must be explicitly specified in `resources` attribute in
+  `<link>` elements, similar to other subresources. `scope` attribute can not be
+  used to specify `urn:uuid` resources.
```

Comment:
> `scope` attribute can not be used to specify `urn:uuid` resources.

- `scope` -> `scopes`
- This does not match the behavior of current Chromium implementation where `scopes="urn:"` allows all `urn:` resources.

Reply:
Thanks! I've updated the PR so that it matches the Chromium implementation.